### ISSUE TITLE
Not compatible with the latest version of dry-struct

### ISSUE THREAD

**Issue by pcriv** [tbuehlmann/dry-struct-setters#1]
Currently setters are generated but they don't update the attributes.

**Comment by tbuehlmann** [comment]
What version of dry-struct do you use? Do you have example code to reproduce the problem?

**Comment by pcriv** [comment]
Version `0.5.0`

I ended up implementing this class:

```ruby
class StructWithSetters < Dry::Struct
  # Convert string keys to symbols
  transform_keys(&:to_sym)

  # Resolve default types on nil
  transform_types do |type|
    if type.default?
      type.constructor do |value|
        value.nil? ? Dry::Types::Undefined : value
      end
    else
      # Make all types omittable
      type.meta(omittable: true)
    end
  end

  def self.attribute(name, type = nil, &block)
    super
    define_attribute_setter(name)
  end

  def self.define_attribute_setter(name)
    define_method("#{name}=") do |value|
      self.attributes = attributes.merge(name => value)
    end
  end

  def attributes=(new_attributes)
    @attributes = self.class.input[new_attributes]
  end
end
```

**Comment by tbuehlmann** [comment]
So this is your solution without using dry-struct-setters, do I understand that correctly?

**Comment by pcriv** [comment]
Yes indeed, i thought to share it in case you want to update this gem.

**Comment by tbuehlmann** [comment]
I first need to reproduce the error. Can you provide a minimal example of what is not working?

**Comment by pcriv** [comment]
```ruby
require "dry/struct/setters"

class User < Dry::Struct
  include Dry::Struct::Setters
  include Dry::Struct::Setters::MassAssignment

  attribute :name, Dry::Types["string"]
end

user = User.new(name: "Pablo")
user.name = "Joe"
=> "Joe"
user.name # => Pablo
```

**Comment by tbuehlmann** [comment]
Great, thanks. Will have a look at it.

**Comment by tbuehlmann** [comment]
I released version 0.2.0 which includes a fix. Wanna give this a shot?

**Comment by tbuehlmann** [comment]
Closing this for inactivity. If the problem persists, feel free to comment or open a new issue.
